acquire/dynamic/windows/handles.py

```diff
@@ -0,0 +1,255 @@
+            return None
+        else:
+            raise RuntimeError(hex(result))
+
+
+def _get_file_name_thread(h_file: HANDLE, q: Queue):
+    iob = IO_STATUS_BLOCK()
+    file_name_information = ctypes.create_string_buffer(0x1000)
+
+    result = ntdll.NtQueryInformationFile(
+        h_file,
+        ctypes.byref(iob),
+        file_name_information,
+        len(file_name_information),
+        FILE_INFORMATION_CLASS.FileNameInformation,
+    )
+
+    file_name = None
+    if result == NtStatusCode.STATUS_SUCCESS:
+        file_name_length = struct.unpack("<I", file_name_information[:4])[0]
+        file_name = file_name_information[4 : 4 + file_name_length].decode("utf-16-le")
+
+    q.put(file_name)
+
+
+def get_handle_name(pid: int, handle: SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX) -> Optional[str]:
+    """Return handle name."""
+
+    remote = pid != kernel32.GetCurrentProcessId()
+
+    if remote:
+        try:
+            h_remote = open_process(pid)
+        except OpenProcessError as e:
+            log.error(e)
+            return None
+        try:
+            handle = duplicate_handle(h_remote, handle)
+        except RuntimeError:
+            close_handle(h_remote)
+            return None
+
+    q = Queue()
+    thread = threading.Thread(target=_get_file_name_thread, args=(handle, q))
+    thread.daemon = True
+    thread.start()
+    thread.join(1.0)
+
+    result = None
+    if not thread.is_alive():
+        try:
+            result = q.get_nowait()
+        except Exception:  # noqa
+            pass
+
+    return result
+
+
+def get_handles() -> Iterable[Handle]:
+    """Returns all handles of a target."""
+    system_handle_information = SYSTEM_HANDLE_INFORMATION_EX()
+    size = DWORD(ctypes.sizeof(system_handle_information))
+    duplicate_filter = DuplicateFilter()
+    log.addFilter(duplicate_filter)
+
+    while True:
+        result = ntdll.NtQuerySystemInformation(
+            SYSTEM_INFORMATION_CLASS.SystemExtendedHandleInformation,
+            ctypes.byref(system_handle_information),
+            size,
+            ctypes.byref(size),
+        )
+
+        if result == NtStatusCode.STATUS_SUCCESS:
+            break
+        elif result == NtStatusCode.STATUS_INFO_LENGTH_MISMATCH:
+            size = DWORD(size.value * 4)
+            ctypes.resize(system_handle_information, size.value)
+        else:
+            raise RuntimeError(hex(result))
+
+    p_handles = ctypes.cast(
+        system_handle_information.Handles,
+        ctypes.POINTER(SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX * system_handle_information.NumberOfHandles),
+    )
+    for handle in p_handles.contents:
+        handle_type = get_handle_type_info(handle.HandleValue)
+        handle_name = get_handle_name(handle.UniqueProcessId, handle.HandleValue)
+
+        if not handle_name:
+            continue
+
+        yield Handle(handle, handle_type, handle_name)
+    log.removeFilter(duplicate_filter)
+
+
+def open_process(pid: int) -> int:
```

Comment:
When reading through the code it is nicer to have this appear above the function that uses it. Could you move it above `get_handle_name()`?